### PR TITLE
[LLVM][AArch64]Add assembly/disassembly for compare-and-branch  instructions

### PR DESCRIPTION
**Pull request by CarolineConcatto**
This patch adds the assembly/disassembly for the following instructions:

CBB<cc>, CBH<cc>,
CB<cc>(immediate), CB<cc>(register)
CBBLE, CBBLO, CBBLS, CBBLT
CBHLE, CBHLO, CBHLS, CBHLT
CBGE, CBHS, CBLE, CBLS (immediate)
CBLE, CBLO, CBLS, CBLT(register)

According to [1]

[1]https://developer.arm.com/documentation/ddi0602

Co-authored-by: Momchil Velikov momchil.velikov@arm.com
Co-authored-by: Spencer Abson spencer.abson@arm.com

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64InstrFormats.td [hunk 1]
```diff
@@ -12607,6 +12659,89 @@ class MulAccumCPA<bit isSub, string asm>
   let Inst{31} = 0b1;
 }
 
+
+//----------------------------------------------------------------------------
+// 2023 Armv9.6 Extensions
```
Comment (jthackray) [inline llvm/lib/Target/AArch64/AArch64InstrFormats.td:12664]: nit: should say 2024

**Review by jthackray — COMMENTED** [review]
Are additional disassembly tests in `llvm/test/MC/Disassembler/AArch64` required? (maybe they're in a later patch?)

**Comment by momchil-velikov** [comment]
> Are additional disassembly tests in `llvm/test/MC/Disassembler/AArch64` required? (maybe they're in a later patch?)

Good point, I've totally forgotten about these tests!

But, no, dissembly is already tested, for example
```
// RUN: llvm-mc -triple=aarch64 -show-encoding -mattr=+cmpbr < %s \
// RUN:        | sed '/.text/d' | sed 's/.*encoding: //g' \
// RUN:        | llvm-mc -triple=aarch64 -mattr=+cmpbr -disassemble -show-encoding \
// RUN:        | FileCheck %s --check-prefixes=CHECK-ENCODING,CHECK-INST
```

**Comment by CarolineConcatto** [comment]
Hi @jthackray ,
Which tests would you like to add the disassembly tests?

**Comment by Lukacma** [comment]
Now that we have instructions which use cmpbr feature, we should also tests that .arch, .arch_extesion and maybe also .cpu directives work with this feature.

**Comment by jthackray** [comment]
> Hi @jthackray , Which tests would you like to add the disassembly tests?

Momchil has pointed out some clever disassembly tests already present, via a pipe, so maybe those are sufficient?

**Review by jthackray — APPROVED** [review]
LGTM.

**Comment by CarolineConcatto** [comment]
I am going to revert until I solve the problem with the sanitiser

**Pull request merged**